src/main/java/net/neoforged/neoforge/event/AddPackFindersEvent.java

```diff
@@ -72,13 +72,16 @@ public void addPackFinders(ResourceLocation packLocation, PackType packType, Com
         if (getPackType() == packType) {
             IModInfo modInfo = ModList.get().getModContainerById(packLocation.getNamespace()).orElseThrow(() -> new IllegalArgumentException("Mod not found: " + packLocation.getNamespace())).getModInfo();
 
-            var resourcePath = modInfo.getOwningFile().getFile().findResource(packLocation.getPath());
-
             var version = modInfo.getVersion();
 
+            String prefix = packLocation.getPath();
+
             var pack = Pack.readMetaAndCreate(
                     new PackLocationInfo("mod/" + packLocation, packNameDisplay, packSource, Optional.of(new KnownPack("neoforge", "mod/" + packLocation, version.toString()))),
-                    BuiltInPackSource.fromName((path) -> new PathPackResources(path, resourcePath)),
+                    BuiltInPackSource.fromName((locationInfo) -> {
+                        var contents = modInfo.getOwningFile().getFile().getContents();
+                        return new JarContentsPackResources(locationInfo, contents, prefix);
```

Comment:
This, however, is not in the test framework -- I feel like this should use a switch to match.

Reply:
Eh, IDK that it's a big deal -- is there any case where the behavior _isn't_ the same, after all? (It ought to be the same I'd think).